### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by kukulauren** [kukulauren/ml-projects#5]
- [ ] Define baseline model configuration- input size, number of layers, optimizer
- [ ] Set up training loop (loss function, optimizer, learning rate, batch size, etc.)
- [ ] Train and evaluate model with single metric
- [ ] Track experiments using tools - MLflow
- [ ] Visualize training curves (loss/accuracy vs. epochs)
- [ ] Check overfit and underfit and decide next steps
- [ ] Tune hyperparameters (learning rate, batch size, number of layers)
- [ ] Test final model on test set
- [ ] Compare model performance across experiments
- [ ] Choose best-performing model checkpoints
